atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/watch/LockWatchManagerImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.watch;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.palantir.atlasdb.keyvalue.api.watch.LockWatchManager;
+import com.palantir.atlasdb.timelock.api.LockWatchRequest;
+import com.palantir.lock.client.NamespacedConjureLockWatchingService;
+import com.palantir.lock.watch.LockWatchReferences;
+
+public final class LockWatchManagerImpl implements LockWatchManager {
+
+    private final Set<LockWatchReferences.LockWatchReference> lockWatchReferences = new HashSet<>();
+    private final NamespacedConjureLockWatchingService lockWatchingService;
+
+    public LockWatchManagerImpl(NamespacedConjureLockWatchingService lockWatchingService) {
+        this.lockWatchingService = lockWatchingService;
+    }
+
+    @Override
+    public synchronized void registerWatches(Set<LockWatchReferences.LockWatchReference> newLockwatches) {
+        lockWatchReferences.addAll(newLockwatches);
+    }
+
+    public synchronized void reregisterWatches() {
```

Comment:
I think I can just make this have a thread that calls this on timelock once every few seconds and not bother with being clever.

Reply:
Gonna do it